WIP [JENKINS-40665] Pipeline support for Maven SettingsProvider & GlobalSettingsProvider

cyrille-leclerc opened the pull request:
Support Maven Settings Provider and Maven Global Settings Provider for pipeline and make it work with `withMaven(){}`

See
* [JENKINS-40665 Maven SettingsProvider and GlobalSettingsProvider should support Pipeline jobs](https://issues.jenkins-ci.org/browse/JENKINS-40665)
* [JENKINS-39407 Use system default Maven, Maven Settings if not specified](https://issues.jenkins-ci.org/browse/JENKINS-39407)
* PR https://github.com/jenkinsci/jenkins/pull/2678
* PR https://github.com/jenkinsci/pipeline-maven-plugin/pull/31
* PR https://github.com/jenkinsci/config-file-provider-plugin/pull/25

cyrille-leclerc committed:
[JENKINS-40665] Add support for "Run" and thus to Pipeline Jobs to Maven SettingsProvider and GlobalSettingsProvider

cyrille-leclerc commented:
@jglick could you have a look at this PR?

Reference: [JENKINS-39407 Use system default Maven, Maven Settings if not specified](https://issues.jenkins-ci.org/browse/JENKINS-39407)

oleg-nenashev commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,6 +24,26 @@
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
```
I think the newly introduced method should support propagation of non-Runtime exceptions if there is a real need to propagate them

oleg-nenashev commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -21,8 +24,28 @@
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run,  @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
+        throw new UnsupportedOperationException("Maven SettingsProvider " + getClass() + " does not yet support injecting Maven settings in jobs of type " + run.getClass());
```
🐛 The implementation should check `instanceof AbstractBuild` and invoke old method by default. Otherwise it should return null and maybe log warning

oleg-nenashev commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -21,8 +24,28 @@
+    /**
      * Configure maven launcher argument list with adequate settings path. Implementations should be aware that this method might get called multiple times during a build.
-     * 
+     *
```
I would expect this method to be deprecated

oleg-nenashev reviewed (changes requested):
IMHO having two unrelated implementation is a very confusing way of extending API. IMHO the old method needs to be properly deprecated as we do in other such cases.

Another 🐛 is for throwing undocumented Runtime exceptions in new APIs

cyrille-leclerc committed:
Deprecate `supplySettings(abstractBuild, taskListener)` as we have introduced `supplySettings(run, workspace, taskListener)` as recommended by @oleg-nenashev.

cyrille-leclerc commented:
@oleg-nenashev PR updated

Commit:
[JENKINS-40718] Search by build param values in Build history widget

Commit:
[JENKINS-40718] Make search in Build history really case insensitive

Commit:
[JENKINS-40718] Some Java 7 cleanup

Commit:
[JENKINS-40718] Generalize Build to AbstractBuild

oleg-nenashev commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +26,45 @@
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
```
Add `@since TODO`

oleg-nenashev commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +26,45 @@
 public abstract class GlobalSettingsProvider extends AbstractDescribableImpl<GlobalSettingsProvider> implements ExtensionPoint {
 
     /**
+     * <p>
```
It may break the summary line in Javadoc. I would keep the first entry without paragraph tags

oleg-nenashev commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +26,45 @@
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        throw new AbstractMethodError("Class " + getClass() + " must override the new supplySettings overload");
```
I would rather call to the old `supplySettings` method if that method is overridden && the run is AbstractBuild. Otherwise I would print warning and return null. In such case new API won't blow up if somebody invokes it for non-updated providers

oleg-nenashev commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +26,45 @@
-     *            the build to provide the settigns for
+     *            the build to provide the settings for
      * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     * @deprecated use {@link #supplySettings(Run, FilePath, TaskListener)}
```
Without the fix proposed above it would be a wrong guidance since the provider implementations may not support the legacy API

oleg-nenashev commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -21,12 +26,44 @@
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run,  @Nonnull FilePath workspace, @Nonnull TaskListener listener) throws IOException, InterruptedException {
```
same comment about pluggability as above

oleg-nenashev commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -21,12 +26,44 @@
+     *
      * @param build
      * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     * @deprecated use {@link #supplySettings(Run, FilePath, TaskListener)}
```
same comment as above

oleg-nenashev commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -46,11 +83,22 @@ public static SettingsProvider parseSettingsProvider(StaplerRequest req) throws
      * @return the path to the settings.xml
+     * @deprecated directly invoke {@link SettingsProvider#supplySettings(Run, FilePath, TaskListener)}
      */
     public static final FilePath getSettingsFilePath(SettingsProvider settings, AbstractBuild<?, ?> build, TaskListener listener) {
```
Should also have the `@Deprecated` annotation

oleg-nenashev reviewed (changes requested):
It's getting closer, but I still feel that throwing errors from unimplemented new methods is not the best behavior we could have. The tests are also missing.

KostyaSha commented on core/src/main/java/jenkins/mvn/DefaultGlobalSettingsProvider.java:
```diff
@@ -20,7 +22,7 @@ public DefaultGlobalSettingsProvider() {
 
     @Override
-    public FilePath supplySettings(AbstractBuild<?, ?> project, TaskListener listener) {
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
```
Breaks binary compatibility.

oleg-nenashev replied:
@KostyaSha This is a method with `@Override`. So it does not break the binary compatibility

cyrille-leclerc replied:
Thanks for reviewing @KostyaSha. Do you agree with @oleg-nenashev's statement? Do you agree that this PR does not break binary compatibility as this `FilePath supplySettings(AbstractBuild, TaskListener)` method is still declared in the parent class  GlobalSettingsProvider: 
https://github.com/cyrille-leclerc/jenkins/blob/JENKINS-40665/core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java#L57

KostyaSha replied:
Yes, i missed override side.

KostyaSha reviewed (changes requested):
binary compatibility breakage

cyrille-leclerc commented:
@oleg-nenashev  thanks for reviewing, I just had a little baby girl, I will be a bit slower to update the PR according to your comments.
@KostyaSha I'll wait for your reply on the binary compatibility issue

KostyaSha commented:
Lost override, dismissing review

damianszczepanik committed:
Polish translations

cyrille-leclerc committed:
Integrate @oleg-nenashev’s comments: add more backward compatibility and more java docs

cyrille-leclerc commented:
@oleg-nenashev PR updated. The tests would be the last bit.

jglick committed:
Merge pull request #109 from jenkinsci-cert/ATH-NPE-in-computer-toggle

[SECURITY-362] Fix NPE when toggleOffline is performed by anonymous

jglick committed:
Merge branch 'security-stable-1.625' into security-stable-1.642

jglick committed:
Merge branch 'security-stable-1.642' into security-stable-1.651

jglick committed:
Merge branch 'security-stable-1.651' into security-stable-2.7

jglick committed:
Merge branch 'security-stable-2.7' into security-stable-2.19

jglick committed:
Merge branch 'security-stable-2.19' into security-stable-2.32

jglick committed:
Merge branch 'security-stable-2.32' into security-master

jglick committed:
Switching to ACL.as.

jglick committed:
[SECURITY-382] Be more compatible with maven-plugin.

jglick committed:
Rearranged tests now that ConsoleNote.encodeToBytes pays attention to Jenkins.instance.

jglick committed:
Merge pull request #111 from jenkinsci-cert/SECURITY-382-fixup

[SECURITY-382] Be more compatible with maven-plugin

jglick committed:
Merge branch 'security-stable-1.625' into security-stable-1.642

jglick committed:
Merge branch 'security-stable-1.642' into security-stable-1.651

jglick committed:
Merge branch 'security-stable-1.651' into security-stable-2.7

jglick committed:
Using Jenkins.getInstanceOrNull.

jglick committed:
Merge branch 'security-stable-2.7' into security-stable-2.19

jglick committed:
Merge branch 'security-stable-2.19' into security-stable-2.32

jglick committed:
Revert "[SECURITY-382] Pick up a fix to maven-plugin."
This reverts commit 4c1a23d9e1b711388beb6e2ab10569b7afde495a.
No longer needed as of 2527e66dd8f9dd23a2571f8bd8f2918edf6fc0cd.

jglick committed:
Merge branch 'security-stable-2.32' into security-master

Vlatombe committed:
[JENKINS-41511] Don't try to set slaveAgentPort when it is enforced

Vlatombe committed:
[JENKINS-41511] Add a test showing the problem

jglick committed:
[FIXED JENKINS-39402] Cap the number of group headers printed by AccessDeniedException2.

jglick committed:
[FIXED JENKINS-34065] Deleting obsolete pinning UI.

jglick committed:
For some reason, Bulgarian translations included a gratuitous newline between key and value, which core/move-l10n.groovy does not yet grok.

daniel-beck committed:
Merge pull request #1485 from jglick/printStackTrace-JDK-6507809

Print stack trace in logical order

Vlatombe committed:
[JENKINS-41511] Remove obsolete comment

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.43

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

daniel-beck committed:
Noting #1485

jglick committed:
Merge commit 'e612721e0bab49af3965a8c7761b9596ab4a4964' into security-master

daniel-beck committed:
Merge pull request #2728 from jglick/pin-JENKINS-34065

[JENKINS-34065] Deleting obsolete pinning UI

daniel-beck committed:
Merge pull request #2726 from Vlatombe/JENKINS-41511

[JENKINS-41511] Don't try to set slaveAgentPort when it is enforced

daniel-beck committed:
Merge pull request #2717 from damianszczepanik/pl

Polish translations

daniel-beck committed:
Merge pull request #2693 from daniel-beck/JENKINS-26977

[FIX JENKINS-26977] Use project-specific check URL

abayer committed:
Reverting last remnants of #2444 from Messages

jglick committed:
Merge branch 'master' into AccessDeniedException2-header-size-JENKINS-39402

jglick committed:
Since reviewers could not agree on a way to cap group headers, simply omitting them altogether by default.
Test still reproduces the original issue when flag is set on:
… org.eclipse.jetty.server.HttpChannel$CommitCallback failed
WARNING: Commit failed
java.io.IOException: Response header too large
	at org.eclipse.jetty.http.HttpGenerator.generateResponse(HttpGenerator.java:402)
	at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:655)
	at …
	at hudson.security.AccessDeniedHandlerImpl.handle(AccessDeniedHandlerImpl.java:57)
	at …
Caused by: java.nio.BufferOverflowException
	at java.nio.HeapByteBuffer.put(HeapByteBuffer.java:189)
	at java.nio.ByteBuffer.put(ByteBuffer.java:859)
	at org.eclipse.jetty.http.HttpGenerator.putTo(HttpGenerator.java:1087)
	at org.eclipse.jetty.http.HttpGenerator.generateHeaders(HttpGenerator.java:705)
	at org.eclipse.jetty.http.HttpGenerator.generateResponse(HttpGenerator.java:387)
	... 66 more
java.lang.AssertionError: expected:<403> but was:<500>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at hudson.security.AccessDeniedException2Test.youAreInGroupHeaders(AccessDeniedException2Test.java:56)

daniel-beck committed:
Use final releases

daniel-beck committed:
Merge branch 'security-stable-1.625' into security-stable-1.642

daniel-beck committed:
Merge branch 'security-stable-1.642' into security-stable-1.651

daniel-beck committed:
Merge branch 'security-stable-1.651' into security-stable-2.7

daniel-beck committed:
Merge branch 'security-stable-2.7' into security-stable-2.19

daniel-beck committed:
Merge branch 'security-stable-2.19' into security-stable-2.32

daniel-beck committed:
Merge branch 'security-stable-2.32' into security-master

erral committed:
fix eu translations

daniel-beck committed:
Merge pull request #2731 from erral/fix-eu-translation

Fix eu translations

Commit:
[JENKINS-40718] Support for Pipeline jobs

daniel-beck committed:
Add link to future security advisory URL for 2.44

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.44

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

kohsuke committed:
Merge branch 'master' of github.com:jenkinsci/jenkins

daniel-beck committed:
Fix changelog entries for 2.43 and 2.44

Commit:
[JENKINS-40718] Filter out sensitive parameters on search

Commit:
[JENKINS-40718] Consider insensitiveSearch user configuration

Commit:
[JENKINS-40718] Reduce duplication in tests

jglick committed:
Merge branch 'master' into AccessDeniedException2-header-size-JENKINS-39402

jglick committed:
@oleg-nenashev requests a warning about the disabled headers.

jglick committed:
Disable Windows builds until they are fixed.

Commit:
Update findFormItem so it can return checked radio buttons

Commit:
Merge pull request #2733 from jglick/suppress-win-tests

Suppress Windows tests for the moment

cyrille-leclerc commented:
The tests in error don't seem to be related to the content of the PR:

```
Tests in error: 
jenkins.security.Security232Test.commonsCollections1(jenkins.security.Security232Test)
  Run 1: Security232Test.commonsCollections1:85 » Socket Connection reset
  Run 2: Security232Test.commonsCollections1:85 » Socket Connection reset
  Run 3: Security232Test.commonsCollections1:85 » Socket Connection reset
  Run 4: Security232Test.commonsCollections1:85 » Socket Connection reset
  Run 5: Security232Test.commonsCollections1:85 » Socket Connection reset
```

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.45

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

oleg-nenashev committed:
Changelog: Noting 2.45

alvarolobato committed:
[FIX JENKINS-41457] Use BUILD_NOW_TEXT for parameterised jobs

daniel-beck committed:
[FIX JENKINS-41765] Allow groovy CLI command via SSH

daniel-beck committed:
Fix 2.45 changelog

alvarolobato committed:
[FIX JENKINS-41457] Include issue number in the test

jglick committed:
[FIXED JENKINS-16634] Do not fail to write a log file just because something deleted the parent directory.

jglick committed:
[FIXED JENKINS-41825] Display an informative message, rather than a Groovy exception, when View.getItems fails.

daniel-beck committed:
Fix typo

daniel-beck committed:
Add Windows build back

This will allow PRs to resolve failing tests on Windows

daniel-beck committed:
Merge pull request #2738 from jglick/RewindableRotatingFileOutputStream-JENKINS-16634

[JENKINS-16634] Do not fail to write a log file just because something deleted the parent directory

daniel-beck committed:
Merge pull request #2735 from alvarolobato/JENKINS-41757

[FIX JENKINS-41457] Use BUILD_NOW_TEXT for parameterised jobs

daniel-beck committed:
Merge pull request #2722 from jglick/new-stapler

Stapler 1.250

jglick committed:
@daniel-beck’s 56da425 merging #2722 implied that JENKINS-40088 was fixed in 2.42; in fact it is toward 2.46.

KostyaSha committed:
Annotations for FederatedLoginService.

Signed-off-by: Kanstantsin Shautsou <kanstantsin.sha@gmail.com>

daniel-beck committed:
Merge pull request #2739 from jglick/View.getItems-JENKINS-41825

[JENKINS-41825] Display an informative message, rather than a Groovy exception, when View.getItems fails

daniel-beck committed:
Merge pull request #2723 from abayer/jenkins-37590-mk2

[FIXED JENKINS-37590] Return a null property if no parameters

daniel-beck commented:
@cyrille-leclerc 

> don't seem to be related

Known flaky test.

KostyaSha commented:
> we have two for redundancy,

And no quorum

oleg-nenashev commented:
It was my AI to review it. Will do it ASAP

daniel-beck committed:
Merge pull request #2727 from jglick/AccessDeniedException2-header-size-JENKINS-39402

[JENKINS-39402] Cap the number of group headers printed by AccessDeniedException2

stephenc committed:
Usage Statistics needs a section otherwise it can be kind of hard to find

Commit:
[JENKINS-40718] Integration tests for case insensitive search

slide committed:
Fix some tests on Windows.

slide committed:
More fixes for Windows

slide committed:
Fix more tests on Windows

slide committed:
More fixes

slide committed:
More test fixes

slide committed:
Fingerprint tests passing

jglick committed:
[JENKINS-41684] Ensure that PluginManager.dynamicLoad runs as SYSTEM (#2732)

* [FIXED JENKINS-41684] Ensure that PluginManager.dynamicLoad runs as SYSTEM.
Test plugin source:
package test;
import hudson.Plugin;
import jenkins.model.Jenkins;
public class ThePlugin extends Plugin {
    @Override
    public void postInitialize() throws Exception {
        Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
    }
}

* @daniel-beck wants this all reindented.

jsoref committed:
Spelling fixes in changelogs, Javadoc, logs and UI (#2718)

* spelling: abstract

* spelling: about

* spelling: absolute

* spelling: across

* spelling: activity

* spelling: actually

* spelling: addition

* spelling: allocating

* spelling: ambiguous

* spelling: analyzes

* spelling: analysis

* spelling: another

* spelling: appear

* spelling: arbitrary

* spelling: artifact

* spelling: assignment

* spelling: associated

* spelling: augment

* spelling: authentication

* spelling: automatically

* spelling: available

* spelling: availability

* spelling: because

* spelling: background

* spelling: beginning

* spelling: boolean

* spelling: browsers

* spelling: building

* spelling: calculation

* spelling: channel

* spelling: column

* spelling: concatenation

* spelling: connect

* spelling: contribute

* spelling: convert

* spelling: copied

* spelling: couldn't

* spelling: scrambled

* spelling: creates-a

* spelling: curr-entry

* spelling: customize

* spelling: diagnostic

* spelling: contain

* spelling: default

* spelling: delimiter

* spelling: descriptor

* spelling: disambiguates

* spelling: different

* spelling: diligently

* spelling: disabled

* spelling: discovered

* spelling: display

* spelling: doesn't

* spelling: dollar

* spelling: downstream

* spelling: dynamically

* spelling: preemptively

* spelling: encrypt

* spelling: erroneous

* spelling: examine

* spelling: existence

* spelling: value

* spelling: february

* spelling: handling

* spelling: hostname

* spelling: convenient

* spelling: identify

* spelling: implementation

* spelling: incorrect

* spelling: individual

* spelling: initialization

* spelling: initialized

* spelling: inputstream

* spelling: instantiated

* spelling: instantiation

* spelling: intended

* spelling: interpreted

* spelling: interrupted

* spelling: invocations

* spelling: kern

* spelling: localization

* spelling: logger

* spelling: malfunctioning

* spelling: methods

* spelling: monitor

* spelling: mutator

* spelling: multiple

* spelling: object

* spelling: configured

* spelling: optionally

* spelling: option

* spelling: overridden

* spelling: parameterized

* spelling: parent

* spelling: permissions

* spelling: plugin

* spelling: potentially

* spelling: preferable

* spelling: problems like

* spelling: programmatically

* spelling: property

* spelling: reallocate

* spelling: recommended

* spelling: redirected

* spelling: registered

* spelling: reliable

* spelling: remember

* spelling: recurrence

* spelling: repeatable

* spelling: repeated

* spelling: resource

* spelling: retrieve

* spelling: returned

* spelling: revision

* spelling: sandwich

* spelling: separator

* spelling: serialization

* spelling: settings

* spelling: shadow

* spelling: should

* spelling: someone

* spelling: source

* spelling: specified

* spelling: style

* spelling: subversion

* spelling: sufficient

* spelling: supplementary

* spelling: suppressing

* spelling: synchronization

* spelling: synchronized

* spelling: this

* spelling: transitioning

* spelling: termination

* spelling: trying

* spelling: truncatable

* spelling: unknown

* spelling: undeployed

* spelling: unnecessary

* spelling: unparseable

* spelling: update

* spelling: upper

* spelling: verify

* spelling: visible

* spelling: warning

* spelling: we're

* spelling: whitespace

* spelling: wide

* spelling: with

* spelling: workspace

* spelling: yielding

* spelling: to

* spelling: by

* spelling: the

* spelling: hours

oleg-nenashev committed:
Merge branch 'JENKINS-32820' of https://github.com/atcarmo/jenkins into bug/JENKINS-32820

Conflicts:
	core/src/main/java/hudson/WebAppMain.java
	core/src/main/java/hudson/lifecycle/SolarisSMFLifecycle.java
	core/src/main/java/hudson/lifecycle/UnixLifecycle.java

oleg-nenashev committed:
Merge pull request #2744 from stephenc/usage-stats-needs-a-section

[FIX JENKINS-32938] Usage Statistics needs a section otherwise it can be kind of hard to find

oleg-nenashev committed:
Merge pull request #2737 from daniel-beck/JENKINS-41765

[FIX JENKINS-41765] Allow groovy CLI command via SSH

varmenise committed:
[FIX JENKINS-41264] Improve Javadoc of JnlpAgentReceiver (#2725)

oleg-nenashev committed:
[JENKINS-32820] Fix the last merge issues

slide committed:
Fix some tests and skip some tests.

oleg-nenashev reviewed (approved):
Looks good to me 🐝 
Ideally it requires some testing to confirm non-AbstractBuilds do not blow up due to AbstractMethodError with this change.

cyrille-leclerc commented:
@oleg-nenashev thanks, I have tested with:

* config-file-provider-plugin PR https://github.com/jenkinsci/config-file-provider-plugin/pull/25 /  [JENKINS-40665 Maven SettingsProvider and GlobalSettingsProvider should support Pipeline jobs](https://issues.jenkins-ci.org/browse/JENKINS-40665)
* pipeline-maven-plugin PR https://github.com/jenkinsci/pipeline-maven-plugin/pull/15 / [JENKINS-39407 Use system default Maven, Maven Settings if not specified](https://issues.jenkins-ci.org/browse/JENKINS-39407)

oleg-nenashev commented:
@KostyaSha @jglick WDYT?

slide committed:
More ignores

slide committed:
Fix missing "j." before jenkins

varmenise committed:
[FIX JENKINS-41264] followup fixing methods' references in javadoc

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.46

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

daniel-beck committed:
Noting #2722 #2723 #2727 #2732 #2735 #2737 #2738 #2739 #2744

slide committed:
Try /nobreak for timeout command.

jglick committed:
Check for null return values from InstanceIdentityProvider methods.

jglick committed:
[JENKINS-41987] Improved message.

slide committed:
Replace timeout with ping solution (Thanks Windows)

slide committed:
Remove commented code

slide committed:
Update to use EchoCommand from test harness and some formatting.

stephenc committed:
[JENKINS-41899] Pick up fix

daniel-beck committed:
Archive changelog for versions older than 2.0

jglick committed:
@daniel-beck forgot to update since tags when merging #1485.

jglick committed:
Merge branch 'master' of git://github.com/jenkinsci/jenkins

bstick12 committed:
[JENKINS-18734] - Call perform(AbstractBuild build, Launcher launcher, BuildListener listener) if implemented in BuildStep

slide committed:
Fix parsing so that the test can complete.

stephenc committed:
Merge pull request #2750 from stephenc/jenkins-41899

[JENKINS-41899] Update version-number dependency to 1.3

stephenc committed:
Noting merge of JENKINS-41899

slide committed:
This is really just to get another build.

oleg-nenashev committed:
Merge pull request #2747 from varmenise/follow-up-JENKINS-41264

[FIX JENKINS-41264] followup fixing methods' references in javadoc

oleg-nenashev committed:
Update remoting to 3.5

https://github.com/jenkinsci/remoting/edit/master/CHANGELOG.md

Fixed issues:

* [JENKINS-40710](https://issues.jenkins-ci.org/browse/JENKINS-40710) - Match headers case-insensitively in `JnlpAgentEndpointResolver` in order to be compliant with HTTP2 lower-case headers. ([PR #139](https://github.com/jenkinsci/remoting/pull/139), [PR #140](https://github.com/jenkinsci/remoting/pull/140))
* [JENKINS-41513](https://issues.jenkins-ci.org/browse/JENKINS-41513) - Prevent `NullPointerException` in `JnlpAgentEndpointResolver` when receiving a header with `null` name. ([PR #140](https://github.com/jenkinsci/remoting/pull/140))
* [JENKINS-41852](https://issues.jenkins-ci.org/browse/JENKINS-41852) - Fix exported object pinning logic to prevent release due to the integer overflow. ([PR #148](https://github.com/jenkinsci/remoting/pull/148))

Improvements:

* [JENKINS-41730](https://issues.jenkins-ci.org/browse/JENKINS-41730) - Add the new `org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver.ignoreJenkinsAgentProtocolsHeader` property, which allows specifying a custom list of supported protocols instead of the one returned by the Jenkins master. ([PR #146](https://github.com/jenkinsci/remoting/pull/146))
* Print the Filesystem Jar Cache directory location in the error message when this cache directory is not writable. ([PR #143](https://github.com/jenkinsci/remoting/pull/143))
* Replace `MimicException` with the older `ProxyException` when serializing non-serializable exceptions thrown by the remote code. ([PR #141](https://github.com/jenkinsci/remoting/pull/141))
* Use OID of the `ClassLoaderProxy` in error message when the proxy cannot be located in the export table. ([PR #147](https://github.com/jenkinsci/remoting/pull/147))

ydubreuil committed:
[JENKINS-42141] Fix performance issue in code merging Tool installer list

jglick committed:
Security232Test seems flaky.

jglick commented on core/src/main/java/jenkins/mvn/FilePathGlobalSettingsProvider.java:
```diff
@@ -32,34 +37,45 @@ public String getPath() {
+            } else {
+                String targetPath = env.expand(path);
+                if (IOUtils.isAbsolute(targetPath)) {
+                    result = new FilePath(new File(targetPath));
```
Really you want to treat this as a path on master and not the agent? Seems like an unwanted “feature”.

jglick commented on core/src/main/java/jenkins/mvn/FilePathGlobalSettingsProvider.java:
```diff
@@ -32,34 +37,45 @@ public String getPath() {
-            EnvVars env = build.getEnvironment(listener);
-            String targetPath = Util.replaceMacro(this.path, build.getBuildVariableResolver());
-            targetPath = env.expand(targetPath);
+            EnvVars env = run.getEnvironment(listener);
```
Note that for Pipeline this will be the build’s global environment—it will not take into consideration `node`, `withEnv`, etc. If you actually want to use a local environment, you must pass in an `EnvVars`.

jglick commented on core/src/main/java/jenkins/mvn/FilePathSettingsProvider.java:
```diff
@@ -33,32 +38,44 @@ public String getPath() {
+                    return new FilePath(new File(targetPath));
+                } else {
+                    FilePath mrSettings = build.getModuleRoot().child(targetPath);
+                    FilePath wsSettings = build.getWorkspace().child(targetPath);
```
Consolidate code duplicated with `FilePathGlobalSettingsProvider` into a helper function.

jglick commented on core/src/main/java/jenkins/mvn/FilePathGlobalSettingsProvider.java:
```diff
@@ -32,34 +37,45 @@ public String getPath() {
+            return result;
         } catch (Exception e) {
-            throw new IllegalStateException("failed to prepare global settings.xml");
+            throw new IllegalStateException(
```
Delete—I thought the whole reason to allow the method to throw `IOException | InterruptedException` was so that we would need to wrap in a `RuntimeException`.

jglick commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +27,57 @@
+     *
+     * <p>Implementations should
+     * <ul>Be aware that this method might get called multiple times during a build.</ul>
+     * <ul>Implement this method. This class provides a default implementation throwing an {@link UnsupportedOperationException}
```
Uh, no it does not.

jglick commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +27,57 @@
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     * @throws IOException typically occurs when the {@code supplySettings()} implementation accesses to the
```
use `@link` please

jglick commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +27,57 @@
+     * @param run       the build / run to provide the settings for
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
```
use `@code`

jglick commented on core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java:
```diff
@@ -21,13 +27,57 @@
+     * so that implementations have time to adapt.</ul>
+     * </p>
+     *
+     * @param run       the build / run to provide the settings for
```
just “build” suffices to refer casually to a `Run`

jglick commented on core/src/main/java/jenkins/mvn/SettingsProvider.java:
```diff
@@ -65,7 +127,9 @@ public static final FilePath getSettingsFilePath(SettingsProvider settings, Abst
      * @return the path to the settings.xml
+     * @deprecated directly invoke {@link SettingsProvider#supplySettings(Run, FilePath, TaskListener)}
      */
+    @Deprecated
```
Then you should do so from `Maven.perform`, right?

jglick committed:
Merge pull request #2749 from jglick/JnlpSlaveAgentProtocol4-robustness

[FIXED JENKINS-41987] Check for null return values from InstanceIdentityProvider methods

kohsuke committed:
Typo

kohsuke committed:
Merge remote-tracking branch 'kohsuke/pr/21' into typo

oleg-nenashev committed:
Merge pull request #2742 from KostyaSha/annotations

Annotations for FederatedLoginService.

jglick committed:
Merge pull request #2753 from jglick/Security232Test

Security232Test seems flaky

oleg-nenashev committed:
Merge pull request #2715 from daspilker/JENKINS-33358

[JENKINS-33358] update Groovy to 2.4.8

kohsuke committed:
Merge pull request #2755 from jenkinsci/typo

Typo

oleg-nenashev committed:
Merge pull request #2751 from oleg-nenashev/remoting/3.5

[JENKINS-40710, JENKINS-41513, JENKINS-41852, JENKINS-41730] - Update remoting to 3.5

oleg-nenashev committed:
Merge pull request #2746 from oleg-nenashev/bug/JENKINS-32820

[JENKINS-32820, JENKINS-42164] - Windows service restart does not retain the build queue

oleg-nenashev committed:
Merge pull request #2724 from olivergondza/apt-maven-plugin-java8

[FIXED JENKINS-36872] Switch to com.mysema.maven:apt-maven-plugin for Java 8 support

oleg-nenashev committed:
Merge pull request #847 from bstick12/JENKINS-18734

[JENKINS-18734] - Call perform(AbstractBuild build, Launcher launcher, BuildListener listener) if implemented in BuildStep

daniel-beck committed:
Remove changelog from jenkinsci/jenkins repository (#2757)

* Remove changelog from jenkinsci/jenkins repository
* Add tombstone file directing people to jenkins-infra/jenkins.io

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.47

stephenc committed:
[FIXED JENKINS-42194] Do not display a warning when ignoring post-commit hooks

stephenc committed:
[JENKINS-42194] Code review reveals valid point, no schedules is not a warning

daniel-beck committed:
[FIX JENKINS-41864] Add warning for rare dates

Previously, impossible dates caused 100% CPU while Jenkins was
trying to find the previous/next occurrence of the date.

Vlatombe committed:
[FIXED JENKINS-41128] createItem in View not working when posting xml

oleg-nenashev committed:
Merge pull request #2752 from ydubreuil/fix-JENKINS-42141-perf-issue

[JENKINS-42141] Fix performance issue in code merging Tool installer list

slide committed:
Fix ip address

oleg-nenashev committed:
Merge pull request #2734 from clguimanMSFT/validateButton-radio

Update findFormItem so it can return checked radio buttons

pjfanning committed:
[JENKINS-31598] upgrade commons-collections due to CVE against v3.2.1 (#2761)

* [JENKINS-31598] upgrade commons-collections due to CVE against v3.2.1

* Fix broken tests

daniel-beck committed:
Reference redirectors and jenkins.io as much as possible (#2756)

* Reference redirectors and jenkins.io as much as possible

* Something went wrong, so this is a troubleshooting URL

* More specific redirect URLs

oleg-nenashev commented:
@cyrille-leclerc Any feedback regarding https://github.com/jenkinsci/jenkins/pull/2678#discussion_r101819865 ?

@KostyaSha ping, I think your -1 is resolved

KostyaSha commented:
> @KostyaSha ping, I think your -1 is resolved

> Lost override, dismissing review. Though my review isn't shown on bottom.

No blockers from my side

daniel-beck committed:
[JENKINS-41864] More Javadoc, rephrase warning message

daniel-beck committed:
Merge pull request #2760 from Vlatombe/JENKINS-41128

[FIXED JENKINS-41128] createItem in View not working when posting xml

daniel-beck committed:
Merge pull request #2758 from stephenc/jenkins-42194

[FIXED JENKINS-42194] Do not display a warning when no schedules

daniel-beck committed:
Merge pull request #2763 from slide/windows-ci

Fix tests on Windows

daniel-beck committed:
Merge pull request #2759 from daniel-beck/JENKINS-41864

[FIX JENKINS-41864] Add warning for rare dates

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.48

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
Merge branch 'master' of github.com:jenkinsci/jenkins

jglick committed:
Noting that ParameterValue.getValue can on occasion return null.

urkon committed:
Fix JENKINS-41756 - Indecently translated items

Removed false and indecently to slovene translated main menu items.

oleg-nenashev committed:
Merge pull request #2767 from urkon/patch-1

Fix JENKINS-41756 - Indecently translated items

Jimilian committed:
Minor memory optimisation for every build execution

jglick committed:
Improved Javadoc for getValue.

stephenc committed:
[FIXED JENKINS-42443] Make f:select display spinner during AJAX requests

stephenc committed:
[JENKINS-34522] Expose the noun for a item as a task via AlternativeUiTextProvider

stephenc committed:
[JENKINS-34522] Remove obsolete translations

stephenc committed:
[JENKINS-42443] Minor review comments

stephenc committed:
[JENKINS-42443] Oops missed checking this in

stephenc committed:
[JENKINS-42443] Review comments

jglick committed:
Merge pull request #2766 from jglick/NPE-JENKINS-42367

Noting that ParameterValue.getValue can on occasion return null

oleg-nenashev committed:
Merge pull request #2770 from Jimilian/minor_memory_optimisation

Minor memory optimisation in label parsing

slide committed:
Ignore deletedFolder test on Windows

Windows doesn't allow deleting directories that have a file open inside of them, so this test is invalid on windows.

slide committed:
Fix missing semi-colon

daniel-beck committed:
Merge pull request #2774 from slide/patch-1

Ignore deletedFolder test on Windows

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.49

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

jtnord committed:
kill those pesky hudson temp files

daniel-beck committed:
Fix translation tool entered newlines

jtnord committed:
one more for Command files

daniel-beck committed:
Fix relative link in SCM polling admin monitor

daniel-beck committed:
Remove incomplete Danish translation

daniel-beck committed:
Merge pull request #2778 from jtnord/jenkinsTempFiles

kill those pesky hudson temp files

daniel-beck committed:
Merge pull request #2780 from daniel-beck/fix-relative-link

Fix relative link in SCM polling admin monitor

daniel-beck committed:
Merge pull request #2683 from szpak/feature/JENKINS-40718-searchByBuildParams

[JENKINS-40718] Search by build params in build history widget

stephenc committed:
Merge pull request #2772 from stephenc/jenkins-34522

[JENKINS-34522] Expose the noun for a item as a task via AlternativeUiTextProvider

stephenc committed:
Fix @since TODO and @since FIXME tags

oleg-nenashev committed:
Update Windows Agent Installer to 1.7 and WinSW to 2.0.2 (#2765)

### WinSW changes

The update includes many fixes and improvements, the full list is provided in the [WinSW changelog](https://github.com/kohsuke/winsw/blob/master/CHANGELOG.md). There are several issues referenced in Jenkins bugtracker:

* [JENKINS-22692](https://issues.jenkins-ci.org/browse/JENKINS-22692) - Connection reset issues when WinSW gets terminated due to the system shutdown
* [JENKINS-23487](https://issues.jenkins-ci.org/browse/JENKINS-23487)- Support of shared directories in WinSW
* [JENKINS-39231](https://issues.jenkins-ci.org/browse/JENKINS-39231) - Enable Runaway Process Killer by default
* [JENKINS-39237](https://issues.jenkins-ci.org/browse/JENKINS-39237) - Auto-upgrade of JNLP agent versions on the slaves

### Windows Agent Installer changes

* Adapt the default configurations to pick fixes above
* Slave => Agent renaming where possible

### Jenkins core changes

* Modify the configuration template, reference advanced options
* Enable Runaway Process Killer by default

* Update Windows Agent Installer to 1.7

* Remove the obsolete jenkins-slave.xml file from the core.

Now it is within windows-slave-installer

* Use the deployed Snapshot for CI

* Pick the release version of windows-slave-installer-1.7

oleg-nenashev committed:
[FIXES JENKINS-42371] - Update remoting from 3.5 to 3.7 (#2773)

* [FIXES JENKINS-42371] - Update remoting to 3.6

Fixed issues:

* [JENKINS-42371](https://issues.jenkins-ci.org/browse/JENKINS-42371) - Properly close the `URLConnection` when parsing connection arguments from the JNLP file. It was causing a descriptor leak in the case of multiple connection attempts. ([PR #152](https://github.com/jenkinsci/remoting/pull/152))

* Remoting 3.6 has been burned

stephenc committed:
[FIXED JENKINS-42390] Search results were not correctly encoding URL query parameters

stephenc committed:
[FIXED JENKINS-34691] Add the ability for ItemListeners to veto copy operations

stephenc committed:
[JENKINS-34691] Add Javadoc comment

stephenc committed:
Merge pull request #2771 from stephenc/jenkins-42443

[FIXED JENKINS-42443] Make f:select display spinner during AJAX requests

stephenc committed:
Fixing @since tags from merge of JENKINS-42443

oleg-nenashev committed:
[JENKINS-42670] - Fix the potential File descriptor leak in the Windows Service installer

flochlaysebastien committed:
Correct french word mistake

aureliesalles committed:
Update french documentation

Fix of some mistakes in french help documentation.

jglick committed:
Save after calling setSecurityRealm or setAuthorizationStrategy.

jglick committed:
[JENKINS-42556] Improved logging for Queue.

jglick committed:
[JENKINS-42556] Run more system threads as SYSTEM.

oleg-nenashev committed:
Merge pull request #2783 from SebastienFlochlay/frenchMistake

Translations: Correct french word mistake in FileName parameter

oleg-nenashev committed:
Merge pull request #2788 from aureliesalles/documentation

Update french documentation

Jimilian committed:
[JENKINS-42585] - Replace Hashtable by more efficient ConcurrentHashMap (#2769)

stephenc committed:
Merge pull request #2781 from stephenc/jenkins-42390

[FIXED JENKINS-42390] Search results were not correctly encoding URL query parameters

oleg-nenashev committed:
Introduce the pull-request template. (#2784)

* Introduce the pull-request template.

This  change xplicitly sets expectations from pull requests.

Reason:
* Highlight the need in autotests and JIRA issues for bugfixes
* ask PR creators to propose the changelog entries for their changes.
* Provide hints about referencing people

* Move the pull request template to .github

stephenc committed:
Don't wrap the InterruptedException without unwrapping

oleg-nenashev committed:
Update SSHD Core to 1.10.

Changelog:

* [PR #9](https://github.com/jenkinsci/sshd-module/pull/9) - Move SSH server port configuration to security options page.

oleg-nenashev committed:
Merge pull request #2793 from oleg-nenashev/bug/JENKINS-42670-WindowsInstallerLink_leak

[JENKINS-42670] - Fix the potential File descriptor leak in the Windows Service installer

yogeek committed:
French localization: Fix some typos (#2786)

* fix typo

* fix typo

daniel-beck committed:
Merge pull request #2791 from jglick/Queue-logging-JENKINS-42556

[JENKINS-42556] Improved logging for Queue

Commit:
Translations: Update the Russian localization

oleg-nenashev committed:
Merge branch 'master' into translations/russian_bigscoop

Signed-off-by: Oleg Nenashev <o.v.nenashev@gmail.com>

Conflicts:
	core/src/main/resources/jenkins/model/Jenkins/projectRelationship-help_ru.properties
	core/src/main/resources/jenkins/model/Jenkins/systemInfo_ru.properties

oleg-nenashev committed:
Russian translations: Use redirectors

Signed-off-by: Oleg Nenashev <o.v.nenashev@gmail.com>

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.50

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

oleg-nenashev committed:
Merge pull request #2798 from oleg-nenashev/translations/russian_bigscoop

Update Russian Translations

daniel-beck committed:
Merge pull request #2779 from daniel-beck/translation-tool-newlines

Fix for translation tool newlines

jglick committed:
@oleg-nenashev noticed a typo

jglick committed:
Merge pull request #2792 from jglick/SYSTEM-JENKINS-42556

[JENKINS-42556] Run more system threads as SYSTEM

jglick committed:
[JENKINS-42556] Updating since tags for #2792.

jglick committed:
Merge branch 'master' into security-settings-save

jglick committed:
Jenkins.setNumExecutors already called save; redundant to do so again.

jglick committed:
Fix test to avoid saving Jenkins.securityRealm/authorizationStrategy after we rewrote $JENKINS_HOME/config.xml, clobbering it.
https://github.com/jenkinsci/jenkins-test-harness/pull/53 would simplify test code a bit.

jglick committed:
Non-Serializable SecurityRealm’s.

jglick committed:
Non-Serializable AuthorizationStrategy.

jglick committed:
Non-Serializable SecurityRealm, and simplifying with MockAuthorizationStrategy.

jglick committed:
Non-Serializable SecurityRealm’s, consolidated into one implementation.

jglick committed:
Replacing complex and non-Serializable AuthorizationStrategy’s with MockAuthorizationStrategy.

batmat committed:
Clarify PermissionGroup.owner is @Nonnull

Permission.owner is already marked @Nonnull, and
is built using PermissionGroup.owner.

oleg-nenashev commented:
Ready to merge then. Will integrate the changes today or tomorrow

jglick commented:
No it is not ready. I discussed some problems with @cyrille-leclerc and he said he would work on it further.

stephenc committed:
Merge pull request #2782 from stephenc/jenkins-34691

[FIXED JENKINS-34691] Add the ability for ItemListeners to veto copy operations

stephenc committed:
Merge pull request #2794 from stephenc/dont-wrap-interrupt

Don't wrap the InterruptedException without unwrapping

jglick committed:
Comment.

jglick committed:
Merge branch 'master' into security-settings-save

jglick committed:
Depending on what seems to have been timing conditions, connectNodeShouldSucceedWithForce could fail with a cryptic message.
Found an error in the launch log:
java.lang.IllegalStateException: Already connected
	at hudson.slaves.SlaveComputer.setChannel(SlaveComputer.java:567)
	at hudson.slaves.SlaveComputer.setChannel(SlaveComputer.java:390)
	at hudson.slaves.CommandLauncher.launch(CommandLauncher.java:132)
	at hudson.slaves.SlaveComputer$1.call(SlaveComputer.java:262)
I think the slave was being launched in the background, and then the test tried to disconnect and reconnect it.
Perhaps if these actions overlapped in time, the loose locking semantics of SlaveComputer.setChannel could cause the error.

batmat committed:
Merge pull request #2805 from batmat/clarify_PermissionGroup.owner_is_Nonnull

Clarify PermissionGroup.owner is @Nonnull

oleg-nenashev committed:
Merge pull request #2796 from oleg-nenashev/sshd/1.10

Update SSHD Core to 1.10.

ydubreuil committed:
[JENKINS-42191] Enhance CLI HTTP proxy support (#2711)

* Enhance CLI HTTP proxy support

Fix few issues around proxy support in CLI:

* make proxy support work with recent Squid version which reply with HTTP/1.1 even if request is HTTP 1.0
* close the socket connected to the proxy if the connection failed
* output an error message when proxy connection failed
* don't do a reverse DNS lookup, instead use the host string provided in X-Jenkins-CLI-Host headers (we don't know if the DNS resolver on the
proxy will be able to resolve the name correctly, or like us).

* Use stdout to output CLI proxy connection error message

* Use the logger to output the error message, not System.out

* Add a 'verbose' option to the CLI to turn logging on

This should help people diagnosing connection issues with the CLI.

* [JENKINS-42191] Set log level to FINEST when -v is passed

jglick committed:
Merge pull request #2790 from jglick/security-settings-save

Save after calling setSecurityRealm or setAuthorizationStrategy

Commit:
Apply @jglick’s recommendations:
* consolidate duplicated code in SettingsProviderHelper#supplySettings
* cleanup javadoc and exceptions
* backward compatibility mechanism on absolute path resolution with the the system property `jenkins.mvn.SettingsProvider.absolutePathBackwardCompatibility`
* add unit tests and integration tests

Commit:
Apply @jglick’s recommendations:
* consolidate duplicated code in SettingsProviderHelper#supplySettings
* cleanup javadoc and exceptions
* backward compatibility mechanism on absolute path resolution with the the system property `jenkins.mvn.SettingsProvider.absolutePathBackwardCompatibility`
* add unit tests and integration tests

cyrille-leclerc commented:
The conflict is just a spelling fix in javadocs: https://github.com/jenkinsci/jenkins/commit/496703d0fe133445e10c7d8d07fa7afd351c8854#diff-a30a831cee36351bda7e7a615e9b559aL27

cyrille-leclerc commented:
@jglick I have a rebased version of this branch at https://github.com/cyrille-leclerc/jenkins/commits/JENKINS-40665-rebased .
Shall I rebase on this Pull Request? Shall I close this Pull Request and create a new Pull Request with the rebased version?

Jimilian committed:
[JENKINS-42319] - Fix broken compareTo method of Run (#2762)

* JENKINS-42319: Fix broken compareTo method of Run

Previous implementation relied on fact that users always compare run
objects with same parent. But this is not always the case.
So, it was not safe to put runs from different parents to TreeSet, i.e.

* Make comparasion more performance friendly

oleg-nenashev committed:
[JENKINS-42724] - Restore the windows-service/jenkins.xml resource to restore compatibility with windows-slaves 1.2 (#2803)

* [JENKINS-42724] - Restore the jenkins-slave.xml file

Windows Slaves plugin performs a direct access to the resources bundled into the core.
Hence the file removal was a bad idea though I have not seen the issue in automatic tests and ATH.

This change also was a last-minute change in https://github.com/jenkinsci/jenkins/pull/2765/ in order to address suggestions from @daniel-beck, hence I didn't test it properly

* [JENKINS-42724] - Update the Windows Agents plugin dependency to 1.3.1

* [JENKINS-42724]  -Revert the war/pom.xml upgrade

TristanFAURE committed:
[JENKINS-42627] CLI is not translated to French (#2787)

[JENKINS-42627] French translation for CLI

daniel-beck commented:
There's currently no notable "open" review feedback, so rebasing shouldn't break GitHub.  @jglick WDYT?

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.51

cyrille-leclerc commented:
Here is the rebased fork, I don't see how it could cause harm to this PR but I am not very experienced in rebasing PRs: https://github.com/cyrille-leclerc/jenkins/tree/JENKINS-40665-rebased

@jglick side not: I failed to use a DockerRule in my tests, I got the following error (gist attached)

```
java.lang.IllegalArgumentException: Argument for @Nonnull parameter 'primaryView' of hudson/model/AllView.migrateLegacyPrimaryAllViewLocalizedName must not be null
```

https://gist.github.com/cyrille-leclerc/d88f4b157e85ca38afeaad4b77ae16f3

Commit:
Merge branch 'JENKINS-40665' of github.com:cyrille-leclerc/jenkins into JENKINS-40665

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

cyrille-leclerc commented:
The rebase made the review of this PR hard to read. I am preparing a new PR.

cyrille-leclerc commented:
Replaced by https://github.com/jenkinsci/jenkins/pull/2818
